From 6625527dfd497d9420f151886fef73cbe0b2cade Mon Sep 17 00:00:00 2001
From: Gerrit Pape <pape@smarden.org>
Date: Wed, 26 May 2010 11:08:12 +0000
Subject: [PATCH] [ARITH] Fix logical or result value

Another change I'm making to the arith code is making || return 0 or 1
only, matching C, POSIX and other shells.

Apart from the compliance issue, it is also bad to expose implementation
details like the exact meaning of 'noeval' to scripts such that they may
come to depend on them.

Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
(cherry picked from commit ea2cf705d50024b09e00f8f5bf9c666429bccfa5)

Conflicts:

	ChangeLog
---
 src/arith_yacc.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/arith_yacc.c b/src/arith_yacc.c
index 74b95f8..6c5a720 100644
--- a/src/arith_yacc.c
+++ b/src/arith_yacc.c
@@ -253,7 +253,7 @@ static intmax_t or(int token, union yystype *val, int op, int noeval)
 
 	b = or(token, val, yylex(), noeval | !!a);
 
-	return a | b;
+	return a || b;
 }
 
 static intmax_t cond(int token, union yystype *val, int op, int noeval)
-- 
1.6.0.3

